src/reporting.cc

```diff
@@ -159,3 +158,106 @@ void PrintReport(const snowcrash::Report& report,
         PrintAnnotation("warning:", *it, source, isUseLineNumbers);
     }
 }
+
+template <typename T>
```

Comment:
I think this could go to VisitorUtils as it is a generic function working with refract.